obspy/io/seiscomp/inventory.py

```diff
@@ -366,34 +417,32 @@ def _read_channel(inventory_root, cha_element, _ns):
     # obtain the sensorID and link to particular publicID <sensor> element
     # in the inventory base node
     sensor_id = cha_element.get("sensor")
-    if sensor_id is None:
-        sensor_element = None
-    else:
-        sensor_element = inventory_root.find(_ns("*[@publicID='" + sensor_id +
-                                                 "']"))
+    sensor_element = instrumentation_register["sensors"].get(sensor_id)
+
     # obtain the poles and zeros responseID and link to particular
     # <responsePAZ> publicID element in the inventory base node
-    response_element = None
     if (sensor_element is not None and
        sensor_element.get("response") is not None):
+
         response_id = sensor_element.get("response")
         if response_id is not None:
-            # Generic search for publicID
-            search = "*[@publicID='" + response_id + "']"
-            response_element = inventory_root.find(_ns(search))
-            if response_element is None:
-                msg = ("Could not find matching response element for "
-                       "'{}'.".format(response_id))
-                raise obspy.ObsPyException(msg)
+            # Change in v0.10 the way identifiers are delimited (# -> /)
+            if len(response_id.split("#")) == 1:
+                resp_type = response_id.split("/")[0]
+            else:
+                resp_type = response_id.split("#")[0]
```

Comment:
Something doesn't look right here - the variable `resp_type` appears to never be used.